src/encord_active/lib/model_predictions/importers.py

```diff
@@ -55,214 +57,504 @@
-
-
-def import_KITTI_labels(
-    project: EncordProject,
-    data_root: Path,
-    prediction_writer: PredictionWriter,
-    file_name_regex: str = KITTI_FILE_NAME_REGEX,
+def migrate_coco_predictions(
+    project_dir: Path,
+    predictions_json: Path,
+    ontology_mapping: Optional[dict[str, int]] = None,
+    image_mapping: Optional[dict[int, str]] = None,
 ):
+    """
+    Migrate predictions from the COCO Results format into the specified Encord Active project.
+
+    :param project_dir: The Encord Active project directory.
+    :param predictions_json: The JSON file containing the predictions.
+    :param ontology_mapping: The mapping from Encord's ontology object hashes to the ids of the COCO categories.
+        This mapping allows for the conversion of COCO categories to their corresponding Encord ontology objects.
+        It is a dictionary where the keys are the ontology object hashes used in the ontology of the project,
+        and the values are the corresponding names of the label classes. It's possible for two ontology objects
+        to match the same category if they respectively cover the annotations with polygon or bounding box shape
+        of that class.
+        If `ontology_mapping` is not specified, the function will attempt to load the mapping
+        from the `ontology_mapping.json` file located in the predictions directory.
+        If such file doesn't exist, a `FileNotFoundError` will be raised.
+    :param image_mapping: The mapping from the ids of the images in the COCO file to Encord's data unit hashes.
+        This mapping allows to accurately match each image mentioned in the COCO results file with its corresponding
+        data unit in Encord. It is a dictionary where the keys are the ids of the images in the COCO file, and the
+        values are the hashes of the corresponding data units.
+        If `image_mapping` is not specified, the function will attempt to load the mapping
+        from the `image_mapping.json` file located in the predictions directory.
+        If such file doesn't exist, a `FileNotFoundError` will be raised.
+    :return: The migrated predictions in Encord's Prediction class format.
+    """
+    # Obtain the predictions contained in the file
+    coco_results = parse_results(_json_load(predictions_json))
 
-    label_files = [f for f in (data_root / "labels").iterdir() if f.suffix.lower() in [".txt", ".csv"]]
+    # Retrieve the mapping from ontology object hashes to object class ids
+    if ontology_mapping is None:
+        ontology_mapping = _json_load(predictions_json.parent / "ontology_mapping.json")
 
-    # === Prepare ontology lookup === #
-    with (data_root / "ontology_label_map.json").open("r", encoding="utf-8") as f:
-        label_name_map = json.load(f)
+    # Retrieve the mapping from COCO image ids to data unit hashes
+    if image_mapping is None:
+        image_mapping = _json_load(predictions_json.parent / "image_mapping.json")
```

Comment:
`__json_load` raises an error. Should we perhaps handle that?

Reply:
Probably the best approach is to document the exception, as there is no recovery workaround if the files don't exist.